### PR TITLE
[MB-9926] Investigate local missing otel service and transaction names

### PR DESCRIPTION
**Pull request by duncan-truss**
## Description

This PR modifies our Open Telemetry setup to provide a default name for our service (milmove) and adds a route path label to our traces (api requests) to make them queryable.

I found this example using a route tag that seemed to be the information we wanted to supply to traces https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp#example-NewHandler .  We don't have individual handlers though like in the example but adding the information to the trace span with the same key I believe is equivalent in our middleware.

This probably warrants a mymove-docs update to the telemetry instructions, I think Ronak created a chore for that already.  We may be able to wait until things become stable.

## Reviewer Notes

Is there anything you would like reviewers to give additional scrutiny?

## Setup

To bring up the docker containers for sending telemetry logs to:
```sh
make telemetry_docker
```
In another terminal tab to start the milmove server:
```sh
TELEMETRY_ENDPOINT=localhost:55680 TELEMETRY_ENABLED=1 make server_run
```

In another terminal tab start the local frontend:
```sh
make client_run
```

1. Open http://localhost:5601
2. Navigate to milmovelocal:3000 and login as an existing or new user
3. (optional) Edit or create new data triggering some api requests.
4. Open the menu in the top left and select Observability > Overview
5. Under APM select Services
6. The previously unknown service is should now be called `milmove`
7. Under APM > Traces all of the non-prime APIs should be categorized under `server-no-tls` locally.
8. Traces are now queryable by their metadata `label.http_route` key

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
* [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-9926) for this change
* https://transcom.github.io/mymove-docs/docs/tools/telemetry/running-telemetry-locally

## Screenshots
Screen Shot 2021-11-16 at 4 58 42 PM
Screen Shot 2021-11-16 at 5 01 01 PM

### CODE REVIEW

File: pkg/middleware/openapi.go [hunk 1]
```diff
@@ -28,6 +30,7 @@ func OpenAPITracing(api OpenAPIWithContext) func(next http.Handler) http.Handler
 					logger.Warn("Cannot get labeler from context")
 				} else {
 					labeler.Add(semconv.HTTPTargetKey.String(matchedRoute.PathPattern))
```
Comment (duncan-truss) [inline pkg/middleware/openapi.go:32]: This line already existed but my understanding is the labeler is adding this attribute to *metrics* not *traces* and those are different otel components.

File: pkg/middleware/openapi.go [hunk 2]
```diff
@@ -28,6 +30,7 @@ func OpenAPITracing(api OpenAPIWithContext) func(next http.Handler) http.Handler
...
+					sdktrace.SpanFromContext(r.Context()).SetAttributes(semconv.HTTPRouteKey.String(matchedRoute.PathPattern))
```
Comment (duncan-truss) [inline pkg/middleware/openapi.go:33]: My basic understanding is that each trace has at least one root span and that's where we're setting these key values on.  It's possible that the handlers themselves can create more child spans and/or events.

I saw some reference to traces having names themselves but I didn't explore much further.

**Comment by pearl-truss** [comment]
> When navigating to http://localhost:5601/ I get this error message: `{"statusCode":503,"error":"Service Unavailable","message":"License is not available."}`

Nevermind! I updating my docker and upping the memory resolved this

**Review by pearl-truss — APPROVED** [review]
Thanks for the clear PR instructions, I was able to see the new milmove labeled service and a list of transactions

**Comment by duncan-truss** [comment]
> Thanks for the clear PR instructions, I was able to see the new milmove labeled service and a list of transactions
> 
> Under the APM services, I see a second service called 'unknown' that doesn't have any activity, is this okay or expected?

@pearl-truss Yay, glad you found the prereq setup docs, it was looking like a network error to me at first glance.

For the unknown service I have seen that and don't know where it's being triggered from.  Maybe it's a misconfigured or unnamed service somewhere in serve.go.  I can play around some more and see if any data ever gets sent to it, or maybe it's just some kind of default?

**Pull request merged**